Client/Frontend/Browser/TabLocationView.swift

```diff
@@ -227,7 +227,12 @@ class TabLocationView: UIView {
             urlTextField.attributedText = attributedString
         } else {
             // If we're unable to highlight the domain, just use the URL as is.
-            urlTextField.text = url?.absoluteString
+            if (url?.host != nil) {
+                let urlHost = Punycode().convertHostFromASCIIToUTF8((url?.host)!)
```

Comment:
The correct pattern here is:

```
if let host = url?.host {
```

but Punycode should probably be an extension on String, so you'd be able to do:

```
if let utf8Host = url?.host?.asciiHostAsUTF8 {
```